src/mavlinkscenario.cpp

```diff
@@ -285,34 +285,42 @@ bool MavlinkScenario::add_onboard_message(const OnboardData &msg) {
      ****************************/    
 
     // timeseries
+    const std::string msg_name = "onboard log/" + msg.get_message_name() + "/";
+    const std::size_t stem_len = msg_name.size();
+    std::string fullname = msg_name;
     for (OnboardData::booldata_t::const_iterator dit = msg.get_booldata().begin(); dit != msg.get_booldata().end(); ++dit) {
         if (dit->first == "t") continue;
-        string fullname = "onboard log/" + msg.get_message_name() + "/" + dit->first;
+        fullname.resize(stem_len);
```

Comment:
Is `resize` followed by `append` any faster? Otherwise I would prefer to just append, to keep the code shorter.